marklogic/datadog_checks/marklogic/check.py

```diff
@@ -0,0 +1,223 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from typing import Any, Dict, Generator, List, Tuple
+
+from requests.exceptions import ConnectionError, HTTPError
+
+from datadog_checks.base import AgentCheck
+
+from .api import MarkLogicApi
+from .config import Config
+from .constants import RESOURCE_METRICS_AVAILABLE, RESOURCE_TYPES, SERVICE_CHECK_RESOURCES
+from .parsers.health import parse_summary_health
+from .parsers.request import parse_per_resource_request_metrics
+from .parsers.resources import parse_resources
+from .parsers.status import (
+    parse_per_resource_status_metrics,
+    parse_summary_status_base_metrics,
+    parse_summary_status_resource_metrics,
+)
+from .parsers.storage import parse_per_resource_storage_metrics, parse_summary_storage_base_metrics
+from .utils import is_resource_included
+
+
+class MarklogicCheck(AgentCheck):
+    __NAMESPACE__ = 'marklogic'
+    SERVICE_CHECK_CONNECT = 'can_connect'
+
+    def __init__(self, *args, **kwargs):
+        # type: (*Any, **Any) -> None
+        super(MarklogicCheck, self).__init__(*args, **kwargs)
+
+        self.config = Config(self.instance)
+        self.api = MarkLogicApi(self.http, self.config.url)
+
+        self.collectors = [
+            self.collect_summary_status_base_metrics,
+            self.collect_summary_status_resource_metrics,
+            self.collect_summary_storage_base_metrics,
+            self.collect_per_resource_metrics,
+        ]
+
+        # Refreshed at each check
+        self.resources_to_monitor = {
+            'forest': [],
+            'database': [],
+            'host': [],
+            'server': [],
+        }  # type: Dict[str, List[Any]]
+        self.resources = []  # type: List[Dict[str, str]]
+
+    def check(self, _):
+        # type: (Any) -> None
+        try:
+            raw_resources = self.api.get_resources()
+        except (HTTPError, ConnectionError) as e:
+            self.warning(
+                "Couldn't connect to URL: %s with exception: %s. Please verify the address is reachable",
+                self.config.url,
+                e,
+            )
+            self.service_check(self.SERVICE_CHECK_CONNECT, self.CRITICAL, self.config.tags)
+            return
```

Comment:
```suggestion
            raise
```

I think we should re-raise here, that way the `agent status` will show a failing integration.